CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -732,6 +732,16 @@ namespace CefSharp
         bool ClientAdapter::GetAuthCredentials(CefRefPtr<CefBrowser> browser, CefRefPtr<CefFrame> frame, bool isProxy,
             const CefString& host, int port, const CefString& realm, const CefString& scheme, CefRefPtr<CefAuthCallback> callback)
         {
+            if (isProxy && CefSharpSettings::Proxy != nullptr && CefSharpSettings::Proxy->IP == StringUtils::ToClr(host) && !String::IsNullOrEmpty(CefSharpSettings::Proxy->Username)
+                && !String::IsNullOrEmpty(CefSharpSettings::Proxy->Password))
+            {
+                auto frameWrapper = gcnew CefFrameWrapper(frame);
```

Comment:
Seems unnecessary to create these wrappers, can just execute the callback directly.